### PR TITLE
Resource scm integration

### LINKED ISSUE #54: Implement Resource "SCM Integration"
https://developers.scalingo.com/scm_integrations

### PR DESCRIPTION
**Pull request by SCedricThomas**
Fix #54

### CODE REVIEW

File: scalingo/resource_scalingo_scm_integration.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		},
+	}
+}
+
+func resourceScmIntegrationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+
+	scmType, _ := d.Get("scm_type").(string)
+	url, _ := d.Get("url").(string)
+	accessToken, _ := d.Get("access_token").(string)
```
Comment (curzolapierre) [inline scalingo/resource_scalingo_scm_integration.go:69]: Nipick: Is there a reason to use `_` as it is not mandatory?

By the way I'd add the `ok` syntax to be sure that the cast is working, WDYT?

File: scalingo/resource_scalingo_scm_integration.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+	integration, err := client.SCMIntegrationsCreate(ctx, scalingo.SCMType(scmType), url, accessToken)
+	if err != nil {
+		return diag.Errorf("fail to add scm integration: %v", err)
```
Comment (curzolapierre) [inline scalingo/resource_scalingo_scm_integration.go:73]:
```suggestion
		return diag.Errorf("fail to create scm integration: %v", err)
```

File: scalingo/resource_scalingo_scm_integration.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	return nil
+}
+
+func resourceScmIntegrationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+
+	id := d.Id()
+	integration, err := client.SCMIntegrationsShow(ctx, id)
+	if err != nil {
+		return diag.Errorf("fail to fetch integration: %v", err)
```
Comment (curzolapierre) [inline scalingo/resource_scalingo_scm_integration.go:86]:
```suggestion
		return diag.Errorf("fail to fetch scm integration: %v", err)
```

**Review by curzolapierre — CHANGES_REQUESTED** [review]
Some comments but seems good.

File: scalingo/resource_scalingo_scm_integration.go (hunk 1 above)
Comment (SCedricThomas) [reply scalingo/resource_scalingo_scm_integration.go:69]: I think it's a good idea ! As this issue is present in the whole repository, do you mind if I fix this in another PR ?

Comment (SCedricThomas) [reply scalingo/resource_scalingo_scm_integration.go:69]: As the value types are checked by the schema itself, this may not be so relevant

Comment (curzolapierre) [reply scalingo/resource_scalingo_scm_integration.go:69]: I agree, so looks good.

**Review by curzolapierre — APPROVED** [review]

**Pull request merged**